### ISSUE TITLE
[Bug]: local variable referenced before assignment

### ISSUE THREAD

**Issue by beliaev-maksim** [juju/python-libjuju#851]
### Description

```
 Traceback (most recent call last):
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/tests/integration/test_charm.py", line 45, in test_relations
    await ops_test.model.deploy(ISTIO_PILOT, channel="1.16/stable", trust=True)
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 1664, in deploy
    resources = await self._add_charmhub_resources(res.app_name,
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/model.py", line 1797, in _add_charmhub_resources
    res = await charm_facade.CharmInfo(entity_url)
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 502, in wrapper
    result = cls.from_json(reply['response'])
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 673, in from_json
    return cls(**d)
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/client/_definitions.py", line 4890, in __init__
    meta_ = CharmMeta.from_json(meta) if meta else None
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 673, in from_json
    return cls(**d)
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/client/_definitions.py", line 5402, in __init__
    assumes_expr_ = ExpressionTree.from_json(assumes_expr) if assumes_expr else None
  File "/home/runner/work/oidc-gatekeeper-operator/oidc-gatekeeper-operator/.tox/integration/lib/python3.8/site-packages/juju/client/facade.py", line 692, in from_json
    cls.splitEntries(v, d)
UnboundLocalError: local variable 'v' referenced before assignment
```

### Urgency

Annoying bug in our test suite

### Python-libjuju version

2.9.42.2

### Juju version

2.9.42

### Reproduce / Test

```python
https://github.com/canonical/oidc-gatekeeper-operator/actions/runs/4983828681/jobs/8921366332?pr=78
```

**Comment by beliaev-maksim** [comment]
IDE also suggests that this is an issue ;)
![image](https://github.com/juju/python-libjuju/assets/51964909/0614c826-7b82-4658-947a-332c049dc6e7)

**Comment by addyess** [comment]
looks like a duplicate of https://github.com/juju/python-libjuju/issues/850

**Comment by cderici** [comment]
> looks like a duplicate of #850

yeah the #854 should fix this as well, @beliaev-maksim it would help if you can confirm 👍

**Comment by cderici** [comment]
fixed by #854
